libfreerdp/codec/h264.c

```diff
@@ -59,6 +59,739 @@ static H264_CONTEXT_SUBSYSTEM g_Subsystem_dummy =
+		offset += h264->iStride[2] * (sys->frameHeight / 2);
+
+		hr = outputBuffer->lpVtbl->Unlock(outputBuffer);
+
+		if (FAILED(hr))
+		{
+			WLog_ERR(TAG, "Unlock failure: 0x%04X", hr);
+			goto error;
+		}
+
+		outputBuffer->lpVtbl->Release(outputBuffer);
+	}
+
+	inputSample->lpVtbl->Release(inputSample);
+
+	return 1;
+
+error:
+	fprintf(stderr, "mf_decompress error\n");
+	return -1;
+}
+
+static int mf_compress(H264_CONTEXT* h264, BYTE** ppDstData, UINT32* pDstSize)
+{
+	H264_CONTEXT_MF* sys = (H264_CONTEXT_MF*) h264->pSystemData;
+
+	return 1;
+}
+
+static void mf_uninit(H264_CONTEXT* h264)
+{
+	H264_CONTEXT_MF* sys = (H264_CONTEXT_MF*) h264->pSystemData;
+
+	if (sys)
+	{
+		if (sys->transform)
+		{
+			sys->transform->lpVtbl->Release(sys->transform);
+			sys->transform = NULL;
+		}
+
+		if (sys->codecApi)
+		{
+			sys->codecApi->lpVtbl->Release(sys->codecApi);
+			sys->codecApi = NULL;
+		}
+
+		if (sys->inputType)
+		{
+			sys->inputType->lpVtbl->Release(sys->inputType);
+			sys->inputType = NULL;
+		}
+
+		if (sys->outputType)
+		{
+			sys->outputType->lpVtbl->Release(sys->outputType);
+			sys->outputType = NULL;
+		}
+
+		if (sys->outputSample)
+		{
+			sys->outputSample->lpVtbl->Release(sys->outputSample);
+			sys->outputSample = NULL;
+		}
+
+		if (sys->mfplat)
+		{
+			FreeLibrary(sys->mfplat);
+			sys->mfplat = NULL;
+		}
+
+		free(h264->pYUVData[0]);
+		h264->pYUVData[0] = h264->pYUVData[1] = h264->pYUVData[2] = NULL;
+		h264->iStride[0] = h264->iStride[1] = h264->iStride[2] = 0;
+
+		sys->MFShutdown();
+		
+		CoUninitialize();
+
+		free(sys);
+		h264->pSystemData = NULL;
+	}
+}
+
+static BOOL mf_init(H264_CONTEXT* h264)
+{
+	HRESULT hr;
+	H264_CONTEXT_MF* sys;
+
+	sys = (H264_CONTEXT_MF*) calloc(1, sizeof(H264_CONTEXT_MF));
+
+	if (!sys)
+		goto error;
+
+	h264->pSystemData = (void*) sys;
+
+	/* http://decklink-sdk-delphi.googlecode.com/svn/trunk/Blackmagic%20DeckLink%20SDK%209.7/Win/Samples/Streaming/StreamingPreview/DecoderMF.cpp */
+
+	sys->mfplat = LoadLibraryA("mfplat.dll");
+
+	if (!sys->mfplat)
+		goto error;
+
+	sys->MFStartup = (pfnMFStartup) GetProcAddress(sys->mfplat, "MFStartup");
+	sys->MFShutdown = (pfnMFShutdown) GetProcAddress(sys->mfplat, "MFShutdown");
+	sys->MFCreateSample = (pfnMFCreateSample) GetProcAddress(sys->mfplat, "MFCreateSample");
+	sys->MFCreateMemoryBuffer = (pfnMFCreateMemoryBuffer) GetProcAddress(sys->mfplat, "MFCreateMemoryBuffer");
+	sys->MFCreateMediaType = (pfnMFCreateMediaType) GetProcAddress(sys->mfplat, "MFCreateMediaType");
+	sys->MFCreateDXGIDeviceManager = (pfnMFCreateDXGIDeviceManager) GetProcAddress(sys->mfplat, "MFCreateDXGIDeviceManager");
+
+	if (!sys->MFStartup || !sys->MFShutdown || !sys->MFCreateSample || !sys->MFCreateMemoryBuffer ||
+		!sys->MFCreateMediaType || !sys->MFCreateDXGIDeviceManager)
+		goto error;
+
+	CoInitializeEx(NULL, COINIT_APARTMENTTHREADED);
+
+	if (h264->Compressor)
+	{
+
```

Comment:
Is that required? If nothing is executed in the if clause why not using if (!h264->Compressor) and only this case?